09-state.md

```diff
@@ -0,0 +1,87 @@
+# Swap State
```

Comment:
Maybe the file name could be 09-swap-state.md, I find state alone a bit too difficult to reason about